lib/ts/claims/primitiveArrayClaim.ts

```diff
@@ -0,0 +1,228 @@
+
+                    const ageInSeconds = (Date.now() - this.getLastFetchedTime(payload, ctx)!) / 1000;
+                    if (maxAgeInSeconds !== undefined && ageInSeconds > maxAgeInSeconds) {
+                        return {
+                            isValid: false,
+                            reason: {
+                                message: "expired",
+                                ageInSeconds,
+                                maxAgeInSeconds
+                            }
+                        };
+                    }
+                    const claimSet = new Set(claimVal);
+                    const isValid = val.every(v => !claimSet.has(v));
+                    return isValid
+                        ? { isValid: isValid }
+                        : {
+                              isValid,
+                              reason: { message: "wrong value", expectedToNotInclude: val, actualValue: claimVal }
+                          };
+                }
+            };
+        },
+        strictEquals: (val: ValueType[], maxAgeInSeconds: number, id?: string): SessionClaimValidator => {
+            return {
+                id: id !== undefined ? id : this.id + "-strictEquals",
+                refresh: ctx => this.refresh(ctx),
+                shouldRefresh: (payload, ctx) =>
+                    this.getValueFromPayload(payload, ctx) === undefined ||
+                    // We know payload[this.id] is defined since the value is not undefined in this branch
+                    (maxAgeInSeconds !== undefined && payload[this.id].t < Date.now() - maxAgeInSeconds * 1000),
+                validate: async (payload, ctx) => {
+                    const claimVal = this.getValueFromPayload(payload, ctx);
+                    if (claimVal === undefined) {
+                        return {
+                            isValid: false,
+                            reason: { message: "value does not exist", expectedValue: val, actualValue: claimVal }
+                        };
+                    }
+
+                    const ageInSeconds = (Date.now() - this.getLastFetchedTime(payload, ctx)!) / 1000;
+                    if (maxAgeInSeconds !== undefined && ageInSeconds > maxAgeInSeconds) {
+                        return {
+                            isValid: false,
+                            reason: {
+                                message: "expired",
+                                ageInSeconds,
+                                maxAgeInSeconds
+                            }
+                        };
+                    }
+                    for (const [i, v] of Array.from(val.entries())) {
+                        if (claimVal[i] !== v) {
+                            return {
+                                isValid: false,
+                                reason: { message: "wrong value", expectedValue: val, actualValue: claimVal }
+                            };
+                        }
+                    }
```

Comment:
Does strictEquals imply that the order also has to be the same? It's ambiguous, therefore maybe we can get rid of this function, and ask users to use something like `includesAll(target) && getValueFromPayload().length === target.length`.

Reply:
It does, so doing `includesAll` + length check would be different. Anyway, there is no use-case for it yet, so I'll just remove it.